src/MongoDB.Driver.Core/Core/Clusters/ICluster.cs

```diff
@@ -66,6 +69,19 @@ public interface ICluster : IDisposable
         /// <returns>A core server session.</returns>
         ICoreServerSession AcquireServerSession();
 
+        /// <summary>
+        /// Gets the crypt client.
+        /// </summary>
+        /// <param name="kmsProviders">The kms providers list.</param>
+        /// <param name="schemaMap">The schema map.</param>
+        /// <param name="useClusterCache">The flag shows whether a cache should be used or not.</param>
+#pragma warning disable CS3002
+        CryptClient GetCryptClient(
+            IReadOnlyDictionary<string, IReadOnlyDictionary<string, object>> kmsProviders,
+            IReadOnlyDictionary<string, BsonDocument> schemaMap = null,
+            bool useClusterCache = true);
```

Comment:
Under what circumstances would we not want to use the cluster cache?

Reply:
It's still under discussion for now. I want to discuss it with @jyemin today.
But currently `useClusterCache = false` is used for `ClientEncryption`. It allows us to dispose `CryptClient` in `ClientEncryption` and do not `dispose` `CryptClient` which is saved in the cluster.

I will come back to this code today.